BuildResidentialScheduleFile/measure.rb

```diff
@@ -0,0 +1,182 @@
+    arg.setDisplayName('Schedules: Random Seed')
+    arg.setUnits('#')
+    arg.setDescription("This numeric field is the seed for the random number generator. Only applies if the schedules type is 'stochastic'.")
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument.makeStringArgument('output_csv_path', true)
+    arg.setDisplayName('Schedules: Path')
+    arg.setDescription('Absolute (or relative) path of the csv file containing user-specified occupancy schedules.')
+    args << arg
+
+    return args
+  end
+
+  # define what happens when the measure is run
+  def run(model, runner, user_arguments)
+    super(model, runner, user_arguments)
+
+    # use the built-in error checking
+    if !runner.validateUserArguments(arguments(model), user_arguments)
+      return false
+    end
+
+    # assign the user inputs to variables
+    args = get_argument_values(runner, arguments(model), user_arguments)
+    args = Hash[args.collect { |k, v| [k.to_sym, v] }]
+
+    hpxml_path = args[:hpxml_path]
+    unless (Pathname.new hpxml_path).absolute?
+      hpxml_path = File.expand_path(File.join(File.dirname(__FILE__), hpxml_path))
+    end
+    unless File.exist?(hpxml_path) && hpxml_path.downcase.end_with?('.xml')
+      fail "'#{hpxml_path}' does not exist or is not an .xml file."
+    end
+
+    hpxml = HPXML.new(hpxml_path: hpxml_path)
+
+    # Create EpwFile object
+    epw_path = hpxml.climate_and_risk_zones.weather_station_epw_filepath
+    if not File.exist? epw_path
+      epw_path = File.join(File.expand_path(File.join(File.dirname(__FILE__), '..', 'weather')), epw_path) # a filename was entered for weather_station_epw_filepath
+    end
+    if not File.exist? epw_path
+      runner.registerError("Could not find EPW file at '#{epw_path}'.")
+      return false
+    end
+    epw_file = OpenStudio::EpwFile.new(epw_path)
+
+    success = create_schedules(runner, hpxml, model, epw_file, args)
+    return false if not success
+
+    # modify the hpxml with the schedules path
+    hpxml.header.schedules_filepath = args[:output_csv_path]
+    hpxml_doc = hpxml.to_oga()
+    XMLHelper.write_file(hpxml_doc, hpxml_path)
+    runner.registerInfo("Wrote file: #{hpxml_path}")
+
+    return true
+  end
+
+  def create_schedules(runner, hpxml, model, epw_file, args)
+    info_msgs = []
+
+    # set the calendar year
+    year_description = model.getYearDescription
+    year_description.setCalendarYear(2007) # default to TMY
+    unless hpxml.header.sim_calendar_year.nil?
+      year_description.setCalendarYear(hpxml.header.sim_calendar_year)
+    end
+    if epw_file.startDateActualYear.is_initialized # AMY
+      year_description.setCalendarYear(epw_file.startDateActualYear.get)
+    end
+
+    # set the timestep
+    timestep = model.getTimestep
+    timestep.setNumberOfTimestepsPerHour(1)
+    unless hpxml.header.timestep.nil?
+      timestep.setNumberOfTimestepsPerHour(60 / hpxml.header.timestep)
+    end
```

Comment:
I'm a bit concerned about updating the model in this measure. How about being safe and creating a new OpenStudio::Model::Model that you make these changes to and then pass into the ScheduleGenerator? This will ensure the active model is unchanged.

Reply:
Even better would be to not pass in the model (or even epw_file?) to the ScheduleGenerator and just pass in the minimal information needed (timestep, leap year, etc.). Then you wouldn't have to make any changes to _any_ model object in this code here. But I don't know how easy that would be.

Reply:
I think we still need to pass in `model` (e.g., `set_vacancy` needs it), but we could get away with not updating `model`.

Reply:
It looks like we are really close to not needing the `model` object at all. I'm going to simplify/refactor some methods so we can avoid it.